packages/stackpress-src/src/plugins/admin/transform/templates/search.ts

```diff
@@ -252,18 +249,49 @@ export default function generate(
   fs: FileSystem
 ) {
   for (const model of registry.model.values()) {
+    //check if the model has at least one searchable field
+    const isSearchable = model.fields.some(field => field.attribute('searchable'));
+    //check if the model has at least one filterable field
+    const isFilterable = model.fields.some(field =>
+      field.attribute('filter.switch') ||
+      field.attribute('filter.text') ||
+      field.attribute('filter.relation')
+    );
+
     const file = path.join(
       directory.getPath(), 
       `${model.name}/admin/templates/search.ink`
     );
     if (!fs.existsSync(path.dirname(file))) {
       fs.mkdirSync(path.dirname(file), { recursive: true });
     }
+
+    const searchBar = isSearchable ? `
+      <if true={results.length > 0}>
+        <form class="flex-grow flex flex-y-center">
+          <field-input border-white class="flex-grow" name="q" value=q />
+          <form-button info padding={10}>
+            <element-icon name="search" />
+          </form-button>
+        </form>
+      </if>
+    ` : '';
+
+    const filterBar = isFilterable ? `
+      <if true={results.length > 0}>
```

Comment:
should be `<if true={results.length > 0 || hasFilters}>`. If no results. It could be because they filtered something. In this case we need to show the filter form so they can adjust the filter. where as:

```js
const hasFilter = Object.values(filters).length > 0;
```